frontend/src/main/scala/bloop/bsp/BloopBspServices.scala

```diff
@@ -114,6 +108,7 @@ final class BloopBspServices(
     .requestAsync(endpoints.BuildTarget.dependencySources)(p => schedule(dependencySources(p)))
     .requestAsync(endpoints.DebugSession.start)(p => schedule(startDebugSession(p)))
     .requestAsync(endpoints.BuildTarget.jvmTestEnvironment)(p => schedule(jvmTestEnvironment(p)))
+    .requestAsync(endpoints.BuildTarget.jvmRunEnvironment)(p => schedule(jvmRunEnvironment(p)))
```

Comment:
Would it be possible reuse the same method here?
```suggestion
     .requestAsync(endpoints.BuildTarget.jvmTestEnvironment)(p => schedule(jvmEnvironment(p)))
    .requestAsync(endpoints.BuildTarget.jvmRunEnvironment)(p => schedule(jvmEnvironment(p)))
```

Reply:
They have different sets of input/output types: JvmTestEnvironmentParams, JvmTestEnvironmentResult etc.

Reply:
I see. Still looks like there is room to reuse more code between `jvmRunEnvironment` and `jvmTestEnvironment`. Something like
```scala
def jvmEnvironment[T](targets: Seq[BuildTargetIdentifier])(fn: Seq[JvmEnvironmentItem] => T): T = ???
```

Reply:
It's a bit unsatisfying to see the duplicate structure for `ifInitialized(None)`/error handling/logging/...